functions/Remove-DbaAgentJob.ps1

```diff
@@ -106,53 +108,48 @@ Removes the job from multiple servers using pipe line
             }
 
             foreach ($j in $Job) {
+                Write-Message -Level Verbose -Message "Processing job $j"
 
-                # Check if the job exists
                 if ($Server.JobServer.Jobs.Name -notcontains $j) {
-                    Write-Message -Message "Job $j doesn't exists on $instance" -Warning
+                    switch ($Mode) {
+                        'Lazy' {
+                            Write-Message -Level Verbose -Message "Job $j doesn't exists on $instance." -Target $instance
+                        }
+                        'Strict' {
+                            Stop-Function -Message "Job $j doesn't exist on $instance." -Continue -ContinueLabel main -Target $instance -Category InvalidData
+                        }
+                    }
                 }
                 else {
-                    # Get the job
-                    try {
-                        $currentjob = $Server.JobServer.Jobs[$j]
-                    }
-                    catch {
-                        Stop-Function -Message "Something went wrong creating the job" -Target $instance -ErrorRecord $_ -Continue
-                    }
-
-                    # Delete the history
-                    if (-not $KeepHistory) {
-                        Write-Message -Message "Purging job history" -Level Verbose
-                        $currentjob.PurgeHistory()
-                    }
-
-                    # Execute
-                    if ($PSCmdlet.ShouldProcess($instance, "Removing the job on $instance")) {
+                    if ($PSCmdlet.ShouldProcess($instance, "Removing the job on $currentJob")) {
                         try {
-                            Write-Message -Message "Removing the job" -Level Verbose
+                            $currentJob = $Server.JobServer.Jobs[$j]
+                            # Delete the history
+                            if (-not $KeepHistory) {
```

Comment:
I would update this to use `Test-Bound`, an internal function that will handle testing whether the parameter was passed or not in a more stable manner.

So if you want to test for "false" meaning it was not passed in you use `Test-Bound 'KeepHistory -Not' this will return `$true` if the KeepHistory was not used or provided.